assests/javascript/index.js

```diff
@@ -136,6 +151,125 @@ async function updateTaskStatus(id, isDone) {
   }
 }
 
+function handleTitle() {
+  const editTaskIcons = Array.from(document.getElementsByClassName("edit-task-icon"));
+
+  editTaskIcons.forEach((editTaskIcon) => {
+    editTaskIcon.addEventListener("click", (event) => {
+      const id = event.target.parentNode.id;
+      const titleBox = document.getElementById(`title-box-${id}`);
+      const taskTitle = titleBox.innerText;
+
+      convertEditIconToSubmitIcon(id);
+      convertTaskTitleToInput(id, titleBox, taskTitle);
+      editTaskTitle(id, taskTitle);
+    })
+  })
+}
+
+function convertEditIconToSubmitIcon(id) {
+  hideEditIcon(id);
+  showSubmitIcon(id);
+}
+
+function hideEditIcon(id) {
+  const editIcon = document.getElementById(`edit-task-icon-${id}`);
+
+  editIcon.classList.remove("edit-task-icon");
+  editIcon.classList.add("hidden");
+}
+
+function showSubmitIcon(id) {
+  const submitIcon = document.getElementById(`submit-task-icon-${id}`);
+
+  submitIcon.classList.remove("hidden");
+  submitIcon.classList.add("submit-task-icon");
+}
+
+function convertTaskTitleToInput(id, titleBox, taskTitle) {
+  titleBox.innerHTML = `<input class="edit-task-title" id="edit-title-${id}" type="text" value="${taskTitle}"/>
+                        <p id="edit-error-${id}" class="edit-title-error"></p>`;
+}
+
+function editTaskTitle(id, taskTitle) {
+  const submitIcon = document.getElementById(`submit-task-icon-${id}`);
+
+  submitIcon.addEventListener("click", function (event) {
+    const newTaskTitle = document.querySelector('input[type=text]').value;
+    const newTaskTitleIsEmpty = !(newTaskTitle.trim());
+
+    if (taskTitle !== newTaskTitle && newTaskTitleIsEmpty === false) {
+      updateTitle(id, newTaskTitle);
+    }
+    else if (newTaskTitleIsEmpty === true) {
+      showEditEmptyInputError(id);
+    }
+  })
+}
+
+async function showEditEmptyInputError(id) {
+  const editTaskInput = document.getElementById(`edit-title-${id}`);
+  const editTitleError = document.getElementById(`edit-error-${id}`);
+
+  editTaskInput.classList.add("error");
+
+  editTitleError.innerText = "Add title!";
+}
+
+async function updateTitle(id, text) {
+  try {
+    const body = JSON.stringify({ text });
+    const headers = { "Content-Type": "application/json" };
+
+    const response = await fetch(`${baseUrl}/todos/${id}`, {
+      method: "PATCH",
+      body,
+      headers,
+    });
+
+    const { data } = await response.json();
+    updateTaskBox(data)
```

Comment:
This is ok and also fine. But maybe for the sake of simplicity. In any case of update you could just re-render the task list. Again, this is fine.

Reply:
I will ask you more about this